nova_vm/src/engine/bytecode/executable/function_declaration_instantiation.rs

```diff
@@ -0,0 +1,334 @@
+        // ii. If hasDuplicates is true, then
+        if has_duplicates {
+            // 1. Perform ! env.InitializeBinding(paramName, undefined).
+            ctx.exe
+                .add_instruction_with_identifier(Instruction::ResolveBinding, param_name);
+            ctx.exe
+                .add_instruction_with_constant(Instruction::StoreConstant, Value::Undefined);
+            ctx.exe
+                .add_instruction(Instruction::InitializeReferencedBinding);
+        }
+    }
+
+    // 22. If argumentsObjectNeeded is true, then
+    if arguments_object_needed {
+        // a. If strict is true or simpleParameterList is false, then
+        //     i. Let ao be CreateUnmappedArgumentsObject(argumentsList).
+        // b. Else,
+        //     i. NOTE: A mapped argument object is only provided for non-strict functions that don't have a rest parameter, any parameter default value initializers, or any destructured parameters.
+        //     ii. Let ao be CreateMappedArgumentsObject(func, formals, argumentsList, env).
+        // TODO: Currently, we always create an unmapped arguments object, even
+        // in non-strict mode.
+        ctx.exe
+            .add_instruction(Instruction::CreateUnmappedArgumentsObject);
+
+        // c. If strict is true, then
+        if strict {
+            // i. Perform ! env.CreateImmutableBinding("arguments", false).
+            // ii. NOTE: In strict mode code early errors prevent attempting to assign to this binding, so its mutability is not observable.
+            ctx.exe.add_instruction_with_identifier(
+                Instruction::CreateImmutableBinding,
+                BUILTIN_STRING_MEMORY.arguments,
+            );
+        } else {
+            // d. Else,
+            //   i. Perform ! env.CreateMutableBinding("arguments", false).
+            ctx.exe.add_instruction_with_identifier(
+                Instruction::CreateMutableBinding,
+                BUILTIN_STRING_MEMORY.arguments,
+            );
+        }
+
+        // e. Perform ! env.InitializeBinding("arguments", ao).
+        ctx.exe.add_instruction_with_identifier(
+            Instruction::ResolveBinding,
+            BUILTIN_STRING_MEMORY.arguments,
+        );
+        ctx.exe
+            .add_instruction(Instruction::InitializeReferencedBinding);
+
+        // f. Let parameterBindings be the list-concatenation of parameterNames and « "arguments" ».
+        // NOTE: reusing `parameter_names` for `parameterBindings`.
+        parameter_names.insert("arguments".into());
+    }
+
+    // 24. Let iteratorRecord be CreateListIteratorRecord(argumentsList).
+    // 25. If hasDuplicates is true, then
+    //   a. Perform ? IteratorBindingInitialization of formals with arguments iteratorRecord and undefined.
+    // 26. Else,
+    //   a. Perform ? IteratorBindingInitialization of formals with arguments iteratorRecord and env.
+    if !formals.has_parameter() {
+        ctx.exe.add_instruction(Instruction::IteratorClose)
```

Comment:
**suggestion:** A comment about how this is here to remove the arguments iterator from the iterator stack would not be out of place perhaps.

Reply:
Done.